prow/cmd/deck/validate-static-files.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# check-file ensures that all static links are valid
+check-file() {
+  for path in $(grep -h -o -E '/static[^"]+' $1); do
```

Comment:
Do you need a `-r` here, seems like needed for grepping `/static` directory

Reply:
No, it is grepping `$1`, looking for strings that start with `/static`, such as `/static/prow_bundle.min.js` that it couldn't find previously.

Reply:
`$1` is from https://github.com/kubernetes/test-infra/pull/20790/commits/87fea4ce8dabcbfef39fb009f3e99ff4c48601f7#diff-4dbb2eb4801e310971dae75e09a6bd4c65a171b48b7f317ac1722ac0cb7ef44cR51-R52, the second one is the absolute path of `//prow/cmd/deck/static`, which is a directory. I have tried this command on my local machine and it complained:

```
$ grep -h -o -E '/static[^"]+' ./prow/cmd/deck/static
grep: ./prow/cmd/deck/static: Is a directory
```

Does bazel do something differently?

Reply:
The `locations` function in the `$(locations //prow/cmd/deck/static)` line you are referring to translates the `//prow/cmd/deck/static` label into the list of paths that label contains.

See https://docs.bazel.build/versions/master/be/general.html#genrule_examples for some documentation on this.

Reply:
Ah. I see, good to learn.

Reply:
Looks like this is the better link: https://docs.bazel.build/versions/master/be/make-variables.html#predefined_label_variables